# Cloud Computing: Exam Part 1

In [20]:
import requests
import pandas as pd
import json

# Task 1 - Aviation Weather Center 

Get information about the airport with the station ID 'EDDH' in JSON-format:

In [21]:
url = 'https://aviationweather.gov/api/data/airport?ids=EDDH&format=json'

params = {'ids': 'EDDH',
          'format': 'json'}
response = requests.get(url) 

if response.status_code == 200:
    airport_resp = response.json()
    print('The API call was successful!')
    display(airport_resp)
else:
    print(f'The API call was not successful! Error Code: {response.status_code}')

The API call was successful!


[{'id': '20137',
  'icaoId': 'EDDH',
  'iataId': 'HAM',
  'faaId': None,
  'name': 'HAMBURG AIRPORT',
  'state': 'HH',
  'country': 'DE',
  'source': 'Intl',
  'type': 'ARP',
  'lat': 53.6318,
  'lon': 9.9951,
  'elev': 8,
  'magdec': '2E',
  'owner': 'P',
  'runways': [{'id': '15/33',
    'dimension': '12024x150',
    'surface': 'Hi',
    'alignment': '150'},
   {'id': '05/23',
    'dimension': '10663x150',
    'surface': 'H',
    'alignment': '048'}],
  'rwyNum': '2',
  'rwyLength': 'L',
  'rwyType': 'H',
  'services': 'S',
  'tower': 'T',
  'beacon': 'B',
  'operations': None,
  'passengers': '11090',
  'freqs': 'ATIS,123.12;TWR,121.27',
  'priority': '2'}]

## 1.1

Print airport name of 'EDDH' with infos from the airport response:

In [22]:
print(f'The name of the airpot with the ID "EDDH" is {airport_resp[0].get('name')}.')

The name of the airpot with the ID "EDDH" is HAMBURG AIRPORT.


**Answer**: The name of the airpot with the ID "EDDH" is HAMBURG AIRPORT.

## 1.2

Print IATA code of 'EDDH' with infos from the airport response:

In [23]:
print(f'The IATA code of the airport with the ID "EDDH" is "{airport_resp[0].get('iataId')}".')

The IATA code of the airport with the ID "EDDH" is "HAM".


**Answer**: The IATA code of the airport with the ID "EDDH" is "HAM".

## 1.3

Print geolocation (longitude/latitude) of 'EDDH' with infos from the airport response:

In [24]:
print(f'The Geolocation of the airport with the ID "EDDH" is: ' + 
      f'longitude: {airport_resp[0].get('lon')}, latitude: {airport_resp[0].get('lat')}.')

The Geolocation of the airport with the ID "EDDH" is: longitude: 9.9951, latitude: 53.6318.


**Answer**: The Geolocation of the airport with the ID "EDDH" is: longitude: 9.9951, latitude: 53.6318.

## 1.4

Print number of runways of 'EDDH' with infos from the airport response:

In [25]:
print(f'The airport with the ID "EDDH" has {len(airport_resp[0].get('runways'))} runways.')

The airport with the ID "EDDH" has 2 runways.


**Answer**: The airport with the ID "EDDH" has 2 runways.